[AI Tutor] CT-432: Get profanity and PII filters working as expected

ebeastlake opened the pull request:
The following PR: 
- Gets the client registering PII and profanity violations from the server
- Modifies the `UserMessage` component to correctly "redact" any flagged messages
- Logs violations, if needed, so we can use the console data to troubleshoot per @dancodedotorg 
- Removes an unneeded constant from the shared constants file

## Links

Jira ticket 1: https://codedotorg.atlassian.net/browse/CT-432
Jira ticket 2: https://codedotorg.atlassian.net/browse/CT-442

## Testing story

Tested manually. 

https://github.com/code-dot-org/code-dot-org/assets/26844240/29dc8a9e-fe7b-408a-b5da-81632d802032

Here's what the log will look like.

<img width="387" alt="Screenshot 2024-04-29 at 10 04 47 PM" src="https://github.com/code-dot-org/code-dot-org/assets/26844240/653b7d2b-5252-4a4b-b63a-c18e80404980">
<img width="369" alt="Screenshot 2024-04-29 at 10 04 57 PM" src="https://github.com/code-dot-org/code-dot-org/assets/26844240/5914c8f3-dad2-4532-b851-f609972f0a93">


## Deployment strategy

## Follow-up work

This style of "redacting" messages is unique to AI Tutor. Gen AI is rendering dismissable "alerts." This is fine for now according to Mark, but I will reach out in the #ai-for-student-learning to continue the conversation.

## PR Checklist:

<!--
  The final step! Before you create your PR, double-check that everything is in order.
  Change [ ] to [X] during creation to check boxes.
-->

- [x] Tests provide adequate coverage
- [x] Privacy and Security impacts have been assessed
- [x] Code is well-commented
- [x] New features are translatable or updates will not break translations
- [x] Relevant documentation has been added or updated
- [x] User impact is well-understood and desirable
- [x] Pull Request is labeled appropriately
- [x] Follow-up work items (including potential tech debt) are tracked and linked

ebeastlake commented on apps/src/aiTutor/chatApi.ts:
```diff
@@ -76,25 +84,25 @@ export async function getChatCompletionMessage(
-    };
+  if (!response) return {status: Status.ERROR};
+
+  switch (response.status) {
```
This logic maps the `response.status` from the existing `ShareFilter` API (which is the type of violation) to one of two non-generic error statuses `profanity_violation` or `pii_violation` we'll store in the db.

The previous checks would never have passed
```
response?.status === Status.PROFANITY_VIOLATION
```
and I switched to a `switch` statement because of the many-1 mapping of ShareFilter statuses to AITutorInteractionStatus.

ebeastlake commented on apps/src/aiTutor/redux/aiTutorRedux.ts:
```diff
@@ -105,7 +104,7 @@ export const askAITutor = createAsyncThunk(
       const assistantChatMessage: ChatCompletionMessage = {
         role: Role.ASSISTANT,
-        status: Status.OK,
+        status: chatApiResponse.status,
```
Storing the `chatApiResponse` status from the server lets us style the messages appropriately later.

ebeastlake commented on apps/src/aiTutor/views/UserMessage.tsx:
```diff
@@ -12,50 +12,47 @@ interface UserMessageProps {
+  'This chat has been hidden because it is inappropriate.';
+const PII_VIOLATION_USER_MESSAGE =
+  'This chat has been hidden because it contains personal information.';
+const ERROR_USER_MESSAGE =
```
I standardized the variable names in this file so they all refer to "PII violation" and "profanity violation" instead of the more generic "inappropriate"/"personal."

ebeastlake commented on apps/src/aiTutor/views/UserMessage.tsx:
```diff
@@ -12,50 +12,47 @@ interface UserMessageProps {
-  return null;
-};
-
-const UserMessage: React.FC<UserMessageProps> = ({message}) => {
```
Breaking this up into a functional component called `displayUserMessage` felt redundant, so I condensed it.

ebeastlake commented on lib/cdo/shared_constants.rb:
```diff
@@ -717,19 +717,12 @@ module SharedConstants
     GENERAL_CHAT: 'general_chat',
   }.freeze
 
-  PII_TYPES = {
```
We don't use these anywhere on the backend, so I removed them. cc @bencodeorg

molly-moen commented on apps/src/aiTutor/chatApi.ts:
```diff
@@ -8,8 +8,25 @@ import {
+// Analogous to https://github.com/code-dot-org/ml-playground/pull/299
+// We want to expose enough information to help troubleshoot false positives
+const logViolationDetails = (response: OpenaiChatCompletionMessage) => {
+  console.info('Violation detected in chat completion response', {
```
should we log this to cloudwatch instead? Or in addition to this?

ebeastlake replied:
Great idea. It'd be nice to not have to wait for users to report these to see how aggressive it's being. Thanks, Molly!

molly-moen replied:
seems reasonable to log a generic event for now!

molly-moen reviewed (approved):
One question on logging, otherwise looks good!

ebeastlake commented:
@molly-moen I added a generic metric so we can view frequency, but I'd want to check with Travis before I start logging student PII/profanity to Cloudwatch. Does that seem reasonable to you?

I also removed the use of the Lab2MetricsReporter [per conversation](https://codedotorg.slack.com/archives/C03DBDN67B7/p1714580873668269?thread_ts=1714576982.072169&cid=C03DBDN67B7) with @sanchitmalhotra126. 

Diff: https://github.com/code-dot-org/code-dot-org/pull/58083/commits/efe22e3fe7952efca9b0555aa651193bd60fecf3
